src/coreComponents/physicsSolvers/multiphysics/SinglePhasePoroelasticKernel.hpp

```diff
@@ -0,0 +1,447 @@
+    real64 const biotTimesPressure = m_biotCoefficient * ( m_fluidPressure[k] + m_deltaFluidPressure[k] );
+    totalStress[0] -= biotTimesPressure;
+    totalStress[1] -= biotTimesPressure;
+    totalStress[2] -= biotTimesPressure;
+
+    // Evaluate fluid phase mass content
+    // --- TODO: temporary solution -----------------------------------------------------------------------------------//
+    //           update PoroElastic Constitutive Model (PECM) such that:                                               //
+    //           (  i) the kernel has a view m_porosityOld to the field stored in PECM                                 //
+    //           ( ii) the newPororosity, stored in PECM, is updated as                                                //
+    //                 m_constitutiveUpdate.porosityUpdate( k, q, m_deltaFluidPressure, volStrainIncrement );          //
+    //           (iii) get methods for porosityNew, dPorosity_dFluidPressure, dPorosity_dVolStrainIncrement            //
+    real64 const biotSkeletonModulusInverse = 0.0; //TODO: 1/N = 0 correct only for biotCoefficient = 1                //
+    real64 const volumetricStrainNew = FE_TYPE::symmetricGradientTrace( dNdX, stack.u_local );                         //
+    real64 const volumetricStrainOld = volumetricStrainNew - FE_TYPE::symmetricGradientTrace( dNdX, stack.uhat_local );//
+    real64 const porosityOld = m_poroRef( k ) + m_biotCoefficient * volumetricStrainOld;// +  DeltaPoro                //
+    real64 const dPorosity_dPressure = biotSkeletonModulusInverse;                                                     //
+    real64 const dPorosity_dVolStrainIncrement =  m_biotCoefficient;                                                   //
+                                                                                                                       //
+    GEOSX_ERROR_IF_GT_MSG( fabs( m_biotCoefficient - 1.0 ),                                                            //
+                           1e-10,                                                                                      //
+                           "Correct only for Biot's coefficient equal to 1" );                                         //
+    // --------------------------------------------------------------------------------------------------------------- //
+    real64 const porosityNew = porosityOld
+                               + m_biotCoefficient * (strainIncrement[0] + strainIncrement[1] + strainIncrement[2] )
+                               + biotSkeletonModulusInverse * m_deltaFluidPressure[k];
+
+
+    // Evaluate body force vector
+    real64 bodyForce[3] = { m_gravityVector[0],
+                            m_gravityVector[1],
+                            m_gravityVector[2]};
+    if( m_gravityAcceleration > 0.0 )
+    {
+      real64 mixtureDensity = ( 1.0 - porosityNew ) * m_solidDensity( k, q ) + porosityNew * m_fluidDensity( k, q );
```

Comment:
m_fluidDensity is not a position independant parameter?

Reply:
The solver uses values cell values or quadrature-point values depending on where the quantities of interest are defined without making any choice. Knowing that the flow solvers is based on a cell-centered method (possibly hybridized) allows for less storage in the constitutive models.